### ISSUE TITLE
[FEATURE] AspectFill and AspectFit: extra arguments for alignment? 

### ISSUE THREAD

**Issue by Ziriax** [mono/SkiaSharp#1160]
Currently the `SKRect.AspectFill` and `SKRect.AspectFit` methods always center the resulting rectangle in `this` rectangle.

It feel more flexible to me to add two extra arguments to these methods, with default values `0.5f`, that describe the alignment of the resulting rectangle in the `this` rectangle? 0 means align to left or top,  1 means align to right or bottom, all values in between shift proportionally.

If this is not acceptable, I would propose to enhance the documentation? It currently doesn't mention the centering behavior.

A more orthogonal API approach would be to add `SKRect.Align` methods that align a rectangle in `this` rectangle, using similar `0..1` alignment arguments?

Thanks,
Peter

**Comment by mattleibow** [comment]
Sounds like a useful thing. Are you able to work around this for now?

**Comment by Ziriax** [comment]
Sure, no problem (I use my own C# 2D math library). I was just exploring the API and encountered this. 

I could make a PR if you think this is useful?

**Comment by mattleibow** [comment]
Yes, that will be very helpful. Add any members that you think will be useful, and then we can talk about each one.
